Simulator/Core/Model.cpp

```diff
@@ -177,23 +178,22 @@ void Model::updateHistory()
 /// Get the Connections class object.
 /// @return Pointer to the Connections class object.
 // ToDo: make smart ptr
-shared_ptr<Connections> Model::getConnections() const
+Connections *Model::getConnections() const
```

Comment:
I'm just putting this in a second place; there are more locations where we have "get" methods that return pointers to objects that we don't implement as formal singletons but which really are (we only create one and use a `unique_ptr` to point to it). Should we return a reference, rather than a pointer? I'm not sure...

Reply:
I am more inclined toward returning a reference. Will make that change in the new PR.

Reply:
From Google Style Guide: "Prefer to return by value or, failing that, return by reference. Avoid returning a pointer unless it can be null."

Reply:
OK, resolving this since it's no longer part of this PR.